src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferReader.cs

```diff
@@ -42,7 +44,9 @@ public BufferReader(TSequence buffer)
 
         public Position Position => _currentPosition + _index;
 
-        public ReadOnlySpan<byte> Span => _currentSpan;
+        public ReadOnlySpan<byte> CurrentSegment => _currentSpan;
```

Comment:
Same comment about `Span`. We don't call it Segment in `IMemoryList` why call it segment here.

Reply:
Cause IMemoryList is a list of '''Memory<T>``` instances. ReadOnlyBuffer is a sequence of segments, not sequence of spans.

Reply:
It's a list of things you can get Memory from, same as here, list of things you can get Span from.

Reply:
I think we should call it Span